README.md

```diff
@@ -112,7 +116,38 @@ set :bucket_write_options, {
 }
 ```
 
-See aws-sdk [S3Client.put_object doc](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/S3/Client.html#put_object-instance_method) for all available options.
+#### Object-level options
+
+You can also set write options for files matching specific [patterns](https://ruby-doc.org/core-2.3.0/Dir.html#method-c-glob) using:
+
+```ruby
+set :object_write_options, {
+  'index.html' => { cache_control: 'no-cache' }
+}
+```
+
+or in a more advanced scenario
+
+```ruby
+set :object_write_options, {
+  'assets/**' => { cache_control: 'public, max-age=86400' },
+  'index.html' => { cache_control: 'no-cache' }
+}
+```
+
+**NOTES:**
+
+* `object_write_options` are evaulated **after** `bucket_write_options` and can override them
+* Also the pattern matching for `object_write_options` is evaluated in the order of definition and overrides on match down the chain. For example defining
+
+```ruby
+set :object_write_options, {
+  'assets/my-script.js' => { cache_control: 'no-cache' },
+  'assets/**' => { cache_control: 'public, max-age=86400' }
+}
+```
+
+will set `Cache-Control: public, max-age=86400` header on `assets/my-script.js` as well!
```

Comment:
Good think to write it down in the README. Not sure it would be the expected behavior, but I understand that matching exact match over pattern would be more complex to implement. I think it is clear enough.